src/panels/lovelace/hui-root.ts

```diff
@@ -608,7 +608,18 @@ class HUIRoot extends LitElement {
       return;
     }
 
-    viewIndex = viewIndex === undefined ? 0 : viewIndex;
+    const isVisible = (view) =>
+      Boolean(
+        this._editMode ||
+          view.visible === undefined ||
+          (Array.isArray(view.visible) &&
+            view.visible.some((e) => e.user === this.hass!.user!.id)) ||
+          view.visible === true
+      );
+    viewIndex =
+      viewIndex === undefined || viewIndex === 0
```

Comment:
Why would we try to look it up if `viewIndex` is -? The old check would only guard against undefined. In what scenario would we link to `0` ? And if so, shouldn't that place be updated to not redirect to 0?

Reply:
good point, I'll track that down